scylla-cdc/src/log_reader.rs

```diff
@@ -141,10 +152,9 @@ impl CDCReaderWorker {
                             })
                         })
                         .collect();
-                } else if had_first_generation {
-                    // FIXME: There may be another generation coming in the future with timestamp < end_timestamp
-                    // that could have been missed because of earlier fetching failures.
-                    // More on this here https://github.com/piodul/scylla-cdc-rust/pull/10#discussion_r826865162
+                } else if !check_missed_generation_enabled {
+                    check_missed_generation_enabled = true;
```

Comment:
Can't we call `fetcher.fetch_next_generation` here? Is it necessary to have the `check_missed_generation_enabled` flag and fo the fetch in the next loop iteration?

Adding more flags/variables further complicates the logic of `CDCReaderWorker::run` - which is already quite complicated - so it would be great if we could avoid it.